### PR TITLE
Add support for custom branch for test repository

### PR DESCRIPTION
**Pull request by sathnaga**
Add support for custom branch for test repository,
bydefault, master branch will be used unless provided in
env.conf

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sathnaga** [comment]
```
# git clone https://github.com/sathnaga/tests -b tests_branch_support tests_branch_support 
# cd tests_branch_support/
# ./avocado-setup.py --bootstrap
06:00:42 INFO    : Check for environment
06:00:42 INFO    : Cleaning the Environment
06:00:45 INFO    : Bootstrapping Framework
06:00:45 INFO    : 	1. Creating Avocado Config
06:00:45 INFO    : 	2. Installing Avocado Framework
06:00:49 INFO    : 	3. Cloning/Updating the repo: avocado-misc-tests with branch master under /home/sath/tests_branch_support/tests/avocado-misc-tests

# cd tests/avocado-misc-tests/;git branch;cd -
* master
/home/sath/tests_branch_support
# vi config/wrapper/env.conf 
# git diff
diff --git a/config/wrapper/env.conf b/config/wrapper/env.conf
index 8a6c80f..cde43b5 100644
--- a/config/wrapper/env.conf
+++ b/config/wrapper/env.conf
@@ -10,7 +10,7 @@ optional = [('avocado-framework-plugin-result-html', '')]
 [tests]
 # Usage Examples:
 # ('https://github.com/avocado-framework-tests/avocado-misc-tests.git', '2.0') -- Clones tagged version 2.0 of test repository
-name = [('https://github.com/avocado-framework-tests/avocado-misc-tests.git', '')]
+name = [('https://github.com/avocado-framework-tests/avocado-misc-tests.git', '1.0')]
 
 [deps_centos]
 packages = gcc,python3-devel,python3-setuptools,numactl,policycoreutils-python

# ./avocado-setup.py --bootstrap
06:02:03 INFO    : Check for environment
06:02:03 INFO    : Cleaning the Environment
06:02:05 INFO    : Bootstrapping Framework
06:02:05 INFO    : 	1. Creating Avocado Config
06:02:05 INFO    : 	2. Installing Avocado Framework
06:02:08 INFO    : 	3. Cloning/Updating the repo: avocado-misc-tests with branch 1.0 under /home/sath/tests_branch_support/tests/avocado-misc-tests

# cd tests/avocado-misc-tests/;git branch;cd -
* 1.0
  master
/home/sath/tests_branch_support
# 
```

**Comment by sathnaga** [comment]
@harish-24 @PraveenPenguin @narasimhan-v fyi

File: avocado-setup.py [hunk 1]
```diff
@@ -199,23 +199,24 @@ def need_bootstrap(enable_kvm=False):
...
-    :param repo: repo link
+    :param repo: tuple of repo link and branch(optional)
     :param basepath: base path where the repository has to be downloaded
     """
-    repo_name = repo.split('/')[-1].split('.')[0]
+    if not repo[1]:
+        branch = "master"
+    else:
+        branch = repo[1]
+    cmd_update = "b=%s;git reset --hard && git checkout master && git remote update && (git branch|grep $b||(git checkout $b && git switch -c $b))" % branch
```
Comment (harish-24) [inline avocado-setup.py:209]: why are we using a `-c` here?

Comment (sathnaga) [reply avocado-setup.py:209]: when checkout to tag, it does not create a branch, rather detached from HEAD, so -c will help to create the new branch with the tag, and it works for branches aswell.

Comment (harish-24) [reply avocado-setup.py:209]:
```
[harish@localhost ~]$ git clone https://github.com/avocado-framework/avocado.git
Cloning into 'avocado'...
remote: Enumerating objects: 100, done.
...
Resolving deltas: 100% (31047/31047), done.
[harish@localhost ~]$ cd avocado
[harish@localhost avocado]$ git branch 
* master
[harish@localhost avocado]$ b='new-branch';git reset --hard && git checkout master && git remote update && (git branch|grep $b||(git checkout $b && git switch -c $b))
HEAD is now at ace4c915 Merge remote-tracking branch 'clebergnu/nrunner_std_size_limits'
Already on 'master'
Your branch is up to date with 'origin/master'.
Fetching origin
error: pathspec 'new-branch' did not match any file(s) known to git
[harish@localhost avocado]$ git branch 
* master
```
Seems branch has not been created. Anything I am missing here?

File: avocado-setup.py [hunk 2]
```diff
@@ -199,23 +199,24 @@ def need_bootstrap(enable_kvm=False):
...
+    repo_name = repo[0].split('/')[-1].split('.')[0]
```
Comment (harish-24) [inline avocado-setup.py:210]: can we use `.git` replacing `.` ?

**Review by harish-24 — APPROVED** [review]
LGTM

**Comment by sathnaga** [comment]
Thanks @harish-24

**Pull request merged**